main.nf

```diff
@@ -24,19 +24,35 @@ version = '0.1'
 params.project = false
 params.reads = "data/*{1,2}.fastq.gz"
 params.email = false
-params.star-fusion = true
+params.star_fusion = true
 params.inspector = false
 params.fusioncatcher = true
 params.sensitivity = 'sensitive'
 params.clusterOptions = false
 params.outdir = './results'
 params.fc_extra_options = ''
+
+params.singleEnd = false
 Channel
-    .fromFilePairs( params.reads, size: 2 )
+    .fromFilePairs( params.reads, size: params.singleEnd ? 1 : 2 ) 
     .ifEmpty { exit 1, "Cannot find any reads matching: ${params.reads}" }
     .into { read_files_star_fusion; fusion_inspector_reads; fusioncatcher_reads}
 
 
+// Validate inputs
+if( params.star_fusion_reference && params.star_fusion ){
+    star_fusion_reference = Channel
+        .fromPath(params.star_fusion_reference)
+        .ifEmpty { exit 1, "STAR-fusion reference not found: ${params.star_fusion_reference}" }
+}
+(star_fusion_reference,star_fusion_reference_fusioninspector) = star_fusion_reference.into(2)
```

Comment:
Probably nicer to do this instead:

```groovy
Channel.fromPath(params.star_fusion_reference)
    .ifEmpty { exit 1, "STAR-fusion reference not found: ${params.star_fusion_reference}" }
    .into { star_fusion_reference, star_fusion_reference_fusioninspector }
```

Reply:
Also, I guess that having this line outside of the `if` statement will crash the pipeline if the `if` evaluates to false?

Reply:
Yeah, I think you are right.
But I assume your code will casue iddues as well. Should be no need to have a star fusion reference if we are not running that process!?

Reply:
Sure - I meant that my code would be inside the `if` sorry.